android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java

```diff
@@ -719,9 +751,101 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             setNewChainLayout();
         } else if (mDeleteAccountButton == v) {
             permanentlyDeleteAccount();
+        } else if (mNewCodeWordsButton == v) {
+            generateNewCodeWords();
+        } else if (mNewQrCodeButton == v) {
+            generateNewQrCode();
         }
     }
 
+    void generateNewCodeWords() {
+        getActivity()
+                .runOnUiThread(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+                                String codePhrase = getPureWords();
+                                assert codePhrase != null && !codePhrase.isEmpty();
+                                String timeLimitedWords =
+                                        getBraveSyncWorker()
+                                                .getTimeLimitedWordsFromPure(codePhrase);
```

Comment:
do we need to do that native on UI thread? I would do it before the `runOnUiThread`

Reply:
I can't explain why I posted that into UI thread, as it already was in UI. Thanks for finding, fixed at https://github.com/brave/brave-core/pull/23051/commits/43cafddcff2013f279118eb258965f7406b4b3cb .